ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -170,6 +181,9 @@ public int hashCode() {
 
   public void start() {
     log.info("Starting query with application id: {}", queryApplicationId);
+    if (queryStateListener.isPresent()) {
```

Comment:
replace with `queryStateListener.ifPresent(kafkaStreams::setStateListener);`

Reply:
Made the change.